### DISCUSSION TITLE
9.0.1.18 (stable) can't work for https://bassu.tv/；weeks ago，the old version can work.

### DISCUSSION THREAD

**Discussion by XroyJ** [aclap-dev/video-downloadhelper#669] [Help and Q&A]
### Your question

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "https://bassu.tv/bass-fishing-video/fishing-stick-baits-poppers-kevin-vandam",
  "error": {
    "details": "SideDownload error: : [in#0 @ 0000018c3012bf00] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://vod-adaptive-ak.vimeocdn.com/exp=1716697027~acl=%2F03e63a58-9c51-4066-8c85-066eed6200d7%2F%2A~hmac=22b96032efa9bae90a8accee9857f28306aa305cf0e8bc396dbbd4864d50749c/03e63a58-9c51-4066-8c85-066eed6200d7/sep/video/fb0e5d7d/playlist.m3u8?query_string_ranges=1.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "downloadable_id": "downloadable_5115610308477722",
    "error": "coapp_failure",
    "id": "coapp_failure_ec23b0f0-f8cc-4226-9816-7282cb0b7553"
  }
}

### Addon & CoApp version and details

version: 9.0.1.18
target: google
channel: stable
lang: zh-CN
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"10a660b384904ab8"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "download_directory": "D:\\迅雷下载",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 2
}

**Comment by Mattmjs** [comment]
The vod adaptive url has expired so it's hard to check the problem. If you get another error then share this information again.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user, like Matt.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

I visited the web page in the page_url tag in the original post above.  It's apparently a subscription web site.  But if you click the play button on the player in the web page, it will get you an embedded YouTube video.  Once I clicked the play button a couple of times -- once was not enough -- VDH recognized the video.  VDH generally has no trouble with YouTube videos.  It had no trouble with this video.  It is, apparently, only a 5-minute excerpt of a longer video that is available only if you buy a subscription to the site.  But VDH was quite happy to download this 5-minute teaser, which played just fine, video & audio, in VLC.  There is no problem here.

Since this is a YouTube video & you are using Chrome, you are doomed.  It is well-known that VDH for Chrome does not download YouTube videos.  This is a legal restriction imposed by Google on Chrome extensions that are provided on the Google Chrome web store.  If you really must have this content, switch to Firefox.

**Reply by XroyJ** [reply]
It's weird，I tried firefox，but still can't work.  Windows is 19045.4412; Firefox is 126.0.1 (64 bit);      :(
{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://bassu.tv/bass-fishing-video/kvds-greatest-winning-habits-may-2024",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#0 @ 000001b90397bb40] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://vod-adaptive-ak.vimeocdn.com/exp=1716962202~acl=%2Fce411990-9dd9-47fa-8d4a-7621aad1b447%2F%2A~hmac=e354bc371e18c494f5f13de54e3245ae081c4ae8e6cd93ba806d485a4054eeb9/ce411990-9dd9-47fa-8d4a-7621aad1b447/sep/video/b4d66b59/playlist.m3u8?query_string_ranges=1.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "id": "coapp_failure_707f0989-d918-412e-960f-9926fb0fb8b4",
    "downloadable_id": "downloadable_4277318376360145"
  }
}

**Reply by GrampaWildWilly** [reply]
Read again the environment I am running.  That video, at least its trailer, downloaded for me.  Are you running all the same pieces of software?

403 Forbidden is often evidence of a web site applying stealth DRM.  That is, they aren't formally using DRM, but they have defined their manifests with authentication required.  Of course, they don't publish how you can authenticate.  That's the stealth part.  Without the manifests, downloading is impossible.

403 Forbidden can also be evidence of your not being quick enough to start the VDH download.  YouTube is known to expire its pages after a time.  So when you see a 403, you should ALWAYS reload the web page & try again at least once.  And be quick about it the second time.

BUT

I went to that page again just now & it's showing me the full hour & 20 minutes of this video.  When I first posted above, all I had access to was a 5-minute teaser.  Now, I'm seeing the whole movie.  Something is different.  I have no idea what, but it's not the same as what I originally encountered.  Wait.  I know what.  It's not the same video.  The first URL you posted says something about baits poppers.  The URL you are posting now is about winning habits.  They are not the same video.  Whatever, I was able to download both of them.

Here's what I've got after completing the download:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d0b8d42d-6906-4565-8729-367ed8add972)

I want to draw your attention to the Network Monitor where I have highlighted a json that includes the word player within its name.  This is an object that appears pretty much all the time on YouTube, so it's no surprise I've found one here.  Over on the right side of the Network Monitor you can see part of the URL of one of the available video tracks for this movie.  It's something long & ugly on the site googlevideo.com, again, typical of YouTube content.  But your error message mentions not simply an HLS master manifest (m3u8 file), which is not what you usually find on YouTube, but that manifest is on Vimeo.  I am totally baffled by what you're reporting.

In any case, I downloaded the whole 80 minutes of this video.  Download time was just over 2 minutes & the file is of size 476M, duration 1:19:40, resolution 1280x720.  It played fine in VLC, although I didn't sit & watch it, just sampled it at intervals.  My sampling showed both video & audio throughout.

I have to think you're doing something wrong.  Something.  What, I can't say.  But it's something.  It could be as simple as just getting on the latest versions of things.  Your initial post said you are on CoApp 2.0.19.  At least that is the latest version.  Your most recent post says you are running the latest Firefox.  That is the new update that just came out earlier today.  But you are not running the beta version of VDH for Firefox.  That is always the recommended version, favored over the public one you can get off the Mozilla add-ons web site.  The beta is housed in our Wiki, whose link appears at the top of every page of this forum.

And let's stick to a single video.  Let's work on this last one, since it appears to be one that I can see without registering for the site.  Moving targets just make things more complicated.  Let's try for something stationary.

**Reply by XroyJ** [reply]
Thanks a lot! I'm not familar with network technology. But it seems there's a way which is good to work. I will try the beta version of VDH as your suggested.  appreciated, have a good day!

**Comment by Mattmjs** [comment]
I believe they're trying to download the video that is served from the vod adaptive address. I tried that earlier before it timed out.
I got several minutes of it before stopping download. There was no 403, however it was only the video and no audio.
This on Firefox, 9.0.1.13 stable.

**Reply by GrampaWildWilly** [reply]
Well, you know the deal with YouTube.  They offer tracks, not videos.  VDH offers the convenience of downloading the tracks & merging them into videos for you.  I'm not surprised you got something that had no audio.  Still, YouTube tracks are not usually hosted on Vimeo.  That's what has me baffled.
